scripts/uberenv/packages/mfem/package.py

```diff
@@ -263,7 +263,7 @@ def find_optional_library(name, prefix):
             'MFEM_DEBUG=%s' % yes_no('+debug'),
             # NOTE: env['CXX'] is the spack c++ compiler wrapper. The real
             # compiler is defined by env['SPACK_CXX'].
-            'CXX=%s' % env['CXX'],
+            'CXX=%s' % env['SPACK_CXX'],
```

Comment:
FYI @v-dobrev -- We're temporarily using a local copy of ``spack``'s ``mfem/package.py`` until we get ``axom``'s spack package posted to ``spack``. I noticed that you added the comment above this line to the original file.

@bmhan12 noticed that using ``env['CXX']`` (i.e. the compiler wrapper) produces warnings in our nvcc configurations that are fixed by ``env['SPACK_CXX']`` (i.e. the actual compiler).

@bmhan12 can fill you in on the details, if you'd like.

Reply:
@kennyweiss, this change will break some of the things spack does through the wrapper. For example, the architecture-specific optimization flags, include paths, rpaths, etc. In other words, this is probably not the right way to resolve the issue with the warnings.

Reply:
Thanks @v-dobrev.  
We won't try to merge this upstream.  (Tag: @white238 )

@bmhan12 -- can you please look into another to resolve the warnings?

Reply:
@kennyweiss  @v-dobrev Definitely. The only flags we need to remove/change are the optimization flags.  In my testing, an axom build using a mfem source build with `CXXFLAGS` set to `'-march=ppc64le -mcpu=pwr9'`  reproduces the warnings. 

@white238 Is there anyway to modify these flags? In the spack environment debug output file, I see that they are specified in
`SPACK_TARGET_ARGS='-march=ppc64le -mcpu=pwr9'; export SPACK_TARGET_ARGS`
`